graphistry/pygraphistry.py

```diff
@@ -382,6 +377,11 @@ def refresh(token=None, fail_silent=False):
                 PyGraphistry._is_authenticated = True
             return PyGraphistry.api_token()
         except Exception as e:
+            if PyGraphistry.store_token_creds_in_memory():
+                logger.debug("JWT refresh via creds")
+                logger.debug("2. @PyGraphistry refresh :relogin")
+                return PyGraphistry.relogin()
+
```

Comment:
shouldn't it be `if not fail_silent: ... raise e` ?

Reply:
fixed